processing/src/main/java/io/druid/query/groupby/epinephelinae/BufferArrayGrouper.java

```diff
@@ -72,17 +72,21 @@ static int requiredBufferCapacity(
                                  .mapToInt(AggregatorFactory::getMaxIntermediateSize)
                                  .sum();
 
-    return Integer.BYTES +                                       // key size
-           getUserBufferCapacity(cardinalityWithMissingValue) +  // total used flags size
-           cardinalityWithMissingValue * recordSize;             // total values size
+    return Integer.BYTES +                                           // grouping key size
```

Comment:
I don't understand where it is actually used, I see only where `getUsedFlagBufferCapacity` (for bitset) and `cardinalityWithMissingValue * recordSize` (for records) space is used